src/Service.Tests/Configuration/Telemetry/FileSinkTests.cs

```diff
@@ -0,0 +1,148 @@
+
+        TelemetryOptions _testTelemetryOptions = new(File: new FileSinkOptions(isFileSinkEnabled, fileSinkPath, rollingInterval));
+        _configuration = _configuration with { Runtime = _configuration.Runtime with { Telemetry = _testTelemetryOptions } };
+
+        File.WriteAllText(configFileName, _configuration.ToJson());
+    }
+
+    /// <summary>
+    /// Cleans up the test environment by deleting the runtime config with telemetry options.
+    /// </summary>
+    [TestCleanup]
+    public void CleanUpTelemetryConfig()
+    {
+        if (File.Exists(CONFIG_WITH_TELEMETRY))
+        {
+            File.Delete(CONFIG_WITH_TELEMETRY);
+        }
+
+        if (File.Exists(CONFIG_WITHOUT_TELEMETRY))
+        {
+            File.Delete(CONFIG_WITHOUT_TELEMETRY);
+        }
+    }
+
+    /// <summary>
+    /// Tests if the services are correctly enabled for File Sink.
+    /// </summary>
+    [TestMethod]
+    public void TestFileSinkServicesEnabled()
+    {
+        // Arrange
+        SetUpTelemetryInConfig(CONFIG_WITH_TELEMETRY, true, "/dab-log-test/file-sink-file.txt");
+
+        string[] args = new[]
+        {
+            $"--ConfigFileName={CONFIG_WITH_TELEMETRY}"
+        };
+        using TestServer server = new(Program.CreateWebHostBuilder(args));
+
+        // Additional assertions to check if File Sink is enabled correctly in services
+        IServiceProvider serviceProvider = server.Services;
+        LoggerConfiguration serilogLoggerConfiguration = serviceProvider.GetService<LoggerConfiguration>();
+        Logger serilogLogger = serviceProvider.GetService<Logger>();
+
+        // If serilogLoggerConfiguration and serilogLogger are not null, File Sink is enabled
+        Assert.IsNotNull(serilogLoggerConfiguration, "LoggerConfiguration for Serilog should be registered.");
+        Assert.IsNotNull(serilogLogger, "Logger for Serilog should be registered.");
+    }
+
+    /// <summary>
+    /// Tests if the logs are flushed to the proper path when File Sink is enabled.
+    /// </summary>
+    /// <summary>
+    /// Tests if the logs are flushed to the proper path when File Sink is enabled.
+    /// </summary>
+    [DataTestMethod]
+    [DataRow("file-sink-test-file.txt")]
+    [DataRow("file-sink-test-file.log")]
+    [DataRow("file-sink-test-file.csv")]
+    public async Task TestFileSinkSucceed(string fileName)
+    {
+        // Arrange
+        SetUpTelemetryInConfig(CONFIG_WITH_TELEMETRY, true, fileName, RollingInterval.Infinite);
+
+        string[] args = new[]
+        {
+            $"--ConfigFileName={CONFIG_WITH_TELEMETRY}"
+        };
+        using TestServer server = new(Program.CreateWebHostBuilder(args));
+
+        // Act
+        using (HttpClient client = server.CreateClient())
+        {
+            HttpRequestMessage restRequest = new(HttpMethod.Get, "/api/Book");
+            await client.SendAsync(restRequest);
+        }
+
+        // Assert
+        Assert.IsTrue(File.Exists(fileName));
```

Comment:
In addition to checking for the existence of the file, can you also please check the contents of the file to verify it has the basic Logging information.. anything that is gauranteed to be present like the DAB output name etc.